overlays/hackage-quirks.nix

```diff
@@ -22,6 +22,9 @@ in { haskell-nix = prev.haskell-nix // {
         # Version of of cabal-install in hackage is broken for GHC 8.10.1
         (lib.optionalAttrs (version == "3.2.0.0") {
           packages.cabal-install.src = final.haskell-nix.sources.cabal-32 + "/cabal-install";
+        })
+        (lib.optionalAttrs (builtins.compareVersions version "3.0.0.0" >= 0
+            && builtins.compareVersions version "3.5" < 0) {
```

Comment:
Don’t we have Greater than, ge, le, lt functions? I think that would make this a lot more readable.

Reply:
Maybe I am just old shcool, but I think the `strcmp` rule is ok.  Always compare to 0 that way:

`x op y` <=> `x y op 0`